EXT_BeamPhysics.md

```diff
@@ -0,0 +1,346 @@
+
+- `velocity/`
+  - type: Optional 3-vector *(real)*
+  - description: (`x`, `y`, `z`) velocity vector. Meant to be used for photons where using `momentum` is not appropriate.
+
+- `weight/`
+    - type: Optional *(real)*
+    - description: If macro-particles are being simulated, the `weight` is the total charge of a macro-particle. This is proportional to the number of particles that a macro-particle represents. Also see `charge`.
+
+
+Non Per-Particle Records of the `Particle Group`
+------------------------------------------------
+
+The following possible records of the `Particle Group` are for specifying properties of the entire group of particles.
+
+- `phaseSpaceFirstOrderMoment/`
+  - type: Optional 6-vector *(real)*
+  - description: First order beam moments of `(x, px, y, py, z, pz)`.
+
+- `phaseSpaceSecondOrderMoment/`
+  - type: Optional 6x6-matrix *(real)*
+  - description: Second order beam moments of `(x, px, y, py, z, pz)`.
+
+Particle Record Dataset Attributes
+----------------------------------
+
+The following attributes can be used with any dataset:
+
+- `minValue`:
+  - type: Optional *(real)*
+  - description: Minimum of the data.
+
+- `maxValue`:
+  - type: Optional *(real)*
+  - description: Maximum of the data.
+
+External Mesh Fields Groups
+===========================
+
+The **external mesh field group** is a group for specifying electric and/or magnetic fields, due to a lattice element, at regularly spaced grid points. For example, the fields due to an RF cavity or the fields due to a magnet. Multiple **external mesh field groups** can be defined in a file. The path for a **external mesh field group** is given by the **externalFieldPath** in the file root group:
+- `externalFieldPath`
+  - type: Required if there are external mesh field group(s) *(string)*
+  - description: Base path to the external mesh field groups. Use the **%T** construct if there are multiple meshes present.
+  - example: `/ExternalFieldMesh/%T/`. Base paths to the external fields group, in this case, would be `/ExternalFieldMesh/1/`, etc.
+  - example: `/ExternalFieldMesh/`. In this case there is only one external fields group.
+
+Notes
+-----
+
+- AC fields can be described using complex numbers. The actual field is the real part of
+
+    Z &ast; Exp[-2 pi i f &ast; (t - t0)]
+
+where `Z` is the complex field, `f` is the Oscillation frequency, `t` is the time, and `t0` is a reference time.
+
+Software that writes complex data must use native complex storage for formats that support complex numbers. For formats that do not have native complex support (for example, HDF5), complex numbers are to be stored using subgroups labeled `r` and `i` for real and imaginary respectively. 
```

Comment:
I would change this to:
```suggestion
Software that writes complex data must use native complex storage for formats that support complex numbers.
For formats that do not have native complex support (for example, HDF5), complex numbers are to be stored using value pairs for real and imaginary component and shall follow the `h5py` convention for annotations.
```
Example code:
```py
import h5py as h5                                                                                
import numpy as np                                                                               

x = np.array([1.+2j, 3.+4j, 5.+6j], np.cdouble)                                                  
x                                                                                                
# array([1.+2.j, 3.+4.j, 5.+6.j])
x.dtype                                                                                          
# dtype('complex128')
f = h5.File('complex.h5', 'w')                                                                   
f['/mydata'] = x                                                                                 
f.close()
```

Creates:
```bash
$ h5ls -d -r complex.h5 
/                        Group
/mydata                  Dataset {3}
    Data:
        (0) {1, 2}, {3, 4}, {5, 6}

$ h5dump complex.h5
HDF5 "complex.h5" {
GROUP "/" {
   DATASET "mydata" {
      DATATYPE  H5T_COMPOUND {
         H5T_IEEE_F64LE "r";
         H5T_IEEE_F64LE "i";
      }
      DATASPACE  SIMPLE { ( 3 ) / ( 3 ) }
      DATA {
      (0): {
            1,
            2
         },
      (1): {
            3,
            4
         },
      (2): {
            5,
            6
         }
      }
   }
}
}
```

I think this totally works well and corresponds what I suggest here: https://github.com/openPMD/openPMD-standard/issues/29#issuecomment-566781013